command/agent/http.go

```diff
@@ -584,6 +584,18 @@ func (s *HTTPServer) parseSource(req *http.Request, source *structs.QuerySource)
 	}
 }
 
+// parseMetaFilter is used to parse the ?node-meta=key:value query parameter, used for
+// filtering results to nodes with the given metadata key/value
+func (s *HTTPServer) parseMetaFilter(req *http.Request, key *string, value *string) {
+	if filter, ok := req.URL.Query()["node-meta"]; ok && len(filter) > 0 {
```

Comment:
So for the moment, just give an error message back to be clear that only one filter at a time can be used? That seems like a good idea.